imas/ids_slice.py

```diff
@@ -0,0 +1,572 @@
+              or nested lists/object array representing structure
+
+        Examples:
+            >>> # Get names from identifiers without looping
+            >>> n = edge_profiles.grid_ggd[0].grid_subset[:].identifier.name.values()
+            >>> # Result: ["nodes", "edges", "cells"]
+            >>>
+            >>> # Get 2D array but as list of arrays (default)
+            >>> rho = core_profiles.profiles_1d[:].grid.rho_tor.values()
+            >>> # Result: [ndarray(100,), ndarray(100,), ...] - list of 106 arrays
+            >>>
+            >>> # Get 2D array reshaped to (106, 100)
+            >>> rho = core_profiles.profiles_1d[:].grid.rho_tor.values(reshape=True)
+            >>> # Result: ndarray shape (106, 100)
+            >>>
+            >>> # 3D ions case - returns object array with structure
+            >>> ion_rho = (
+            ...     core_profiles.profiles_1d[:].ion[:].element[:].density.values(
+            ...         reshape=True
+            ...     )
+            ... )
+            >>> # Result: object array shape (106, 3, 2) with IDSNumericArray elements
+        """
+        from imas.ids_primitive import IDSPrimitive, IDSNumericArray
+
+        # Default behavior: return flat list without reshape
+        if not reshape:
+            result = []
+            for element in self._matched_elements:
+                if isinstance(element, IDSPrimitive):
+                    result.append(element.value)
+                else:
+                    result.append(element)
+            return result
+
+        # Multi-dimensional case with reshape requested
+        flat_values = []
+        for element in self._matched_elements:
+            if isinstance(element, IDSPrimitive):
+                flat_values.append(element.value)
+            elif isinstance(element, IDSNumericArray):
+                flat_values.append(
+                    element.data if hasattr(element, "data") else element.value
+                )
+            else:
+                flat_values.append(element)
+
+        # For 1D, just return as is
+        if len(self._virtual_shape) == 1:
+            return flat_values
+
+        # Try to reshape to multi-dimensional shape
+        try:
+            # Calculate total size
+            total_size = 1
+            for dim in self._virtual_shape:
+                total_size *= dim
+
+            # Check if sizes match
+            if len(flat_values) == total_size:
+                # Successfully reshape to multi-dimensional
+                return np.array(flat_values, dtype=object).reshape(self._virtual_shape)
+        except (ValueError, TypeError):
+            pass
+
+        # If reshape fails or not all elements are extractable, return as object array
+        try:
+            return np.array(flat_values, dtype=object).reshape(self._virtual_shape[0:1])
+        except (ValueError, TypeError):
+            return flat_values
+
+    def to_array(self) -> np.ndarray:
```

Comment:
I would also simplify this method: tensorization is only relevant for when the slice refers to leaf nodes (e.g. FLT_1D).

Proposal:
- to_array raises an Exception when the IDSSlice refers to IDSStructure or IDSStructArrays
- to_array raises an Exception when the shape is irregular
- to_array never returns object arrays (when the user calls this method, they want a tensorized array, not object arrays)